python/ray/remote_function.py

```diff
@@ -148,8 +151,13 @@ def _remote(self,
                 memory=None,
                 object_store_memory=None,
                 resources=None,
-                max_retries=None):
+                max_retries=None,
+                internal_called=False):
```

Comment:
Why do we need this `internal_called` flag? Can we just remove all instances of calling this instead?

Reply:
I think we can just get rid of `_remote()` entirely instead. Looks like `.remote()` just calls `._remote()` directly - we can just move the implementation to `.remote()` and leave a deprecation error in `._remote()`